homeassistant/components/recorder/statistics.py

```diff
@@ -1077,6 +1080,52 @@ def get_last_short_term_statistics(
     )
 
 
+def get_latest_short_term_statistics(
+    hass: HomeAssistant, statistic_ids: list[str], convert_units: bool
+) -> dict[str, list[dict]]:
+    """Return the latest short term statistics for statistic_ids."""
+    # This function doesn't use a baked query since its
+    # deprecated since version 1.4
+    with session_scope(hass=hass) as session:
+        # Fetch metadata for the given statistic_id
+        metadata = get_metadata_with_session(hass, session, statistic_ids=statistic_ids)
+        if not metadata:
+            return {}
+        metadata_ids = [
+            metadata[statistic_id][0]
+            for statistic_id in statistic_ids
+            if statistic_id in metadata
+        ]
+        most_recent_metadata_ids = (
```

Comment:
The naming of the variable is confusing, i think it was referring to the id column in tables `Statistics` / `StatisticsShortTerm` when it was called `most_recent_statistic_ids` in `_statistics_at_time`.

Maybe we could rename it to `most_recent_statistic_row` or something, since that's what the query will find, both here and in `_statistics_at_time`?

Reply:
`most_recent_statistic_row` WFM